src/vaadin-message-input.js

```diff
@@ -65,6 +79,29 @@ class MessageInputElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     // Set initial height to one row
     textarea.setAttribute('rows', 1);
     textarea.style.minHeight = '0';
+
+    // Add enter handling for text area.
+    textarea.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter' && !e.shiftKey) {
+        e.preventDefault();
+        e.stopPropagation();
+        this.submitMessage();
+      }
+    });
+  }
+
+  /**
+   * Submits the current value as an custom event named 'submit'.
+   * It also clears the text input and refocuses it for sending another message.
+   * In UI, can be triggered by pressing the submit button or pressing enter key when field is focused.
+   * It does not submit anything if text is empty.
+   */
+  submitMessage() {
```

Comment:
Do we want to add this API? What's the use case?

Reply:
Would a  `_` in front of the function name be enough of a change?

Reply:
IIRC `_` means protected and `__` (two underscores) means private. But I'm not sure actually what protected means in this context. So double underscore is at least the safest. Does @web-padawan have a recommendation?

Reply:
Renamed to `__submit` for now.